examples/advanced-indexing.ipynb

```diff
@@ -0,0 +1,414 @@
+  {
+   "cell_type": "markdown",
+   "id": "d495e16b",
+   "metadata": {},
+   "source": [
+    "## Intro"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "53be532d",
+   "metadata": {},
+   "source": [
+    "Consider extracting a collection of slices of a 3-D shaped like ~ $2 \\times 30 \\times 4$:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "c20ad89c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input = randint(0, 10, (3, 30, 4))\n",
+    "a = input[:, 0, :]\n",
+    "b = input[:, 5, :]\n",
+    "c = input[:, 14, :]\n",
+    "d = input[:, 24, :]\n",
+    "# ..."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "414ab497",
+   "metadata": {},
+   "source": [
+    "This can also be done more easily using \"advanced indexing\":"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "ff7b52f4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "abcd = input[:,(0, 5, 14, 24), :]\n",
+    "assert np.array_equal(abcd[:, 0, :], a)\n",
+    "assert np.array_equal(abcd[:, 1, :], b)\n",
+    "assert np.array_equal(abcd[:, 2, :], c)\n",
+    "assert np.array_equal(abcd[:, 3, :], d)\n",
+    "assert abcd.shape == (3, 4, 4)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5f59a62c",
+   "metadata": {},
+   "source": [
+    "You can also simultaneously reshape the result by provinding the indexes in some shape:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "6d20ddb1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "indices = np.array([[0, 5], [14, 24]])\n",
+    "abcd = input[:, indices, :]\n",
+    "assert np.array_equal(abcd[:, 0, 0, :], a)\n",
+    "assert np.array_equal(abcd[:, 0, 1, :], b)\n",
+    "assert np.array_equal(abcd[:, 1, 0, :], c)\n",
+    "assert np.array_equal(abcd[:, 1, 1, :], d)\n",
+    "assert abcd.shape == (3, 2, 2, 4)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e3e1bd7c",
+   "metadata": {},
+   "source": [
+    "Notice that the result is expanded to accomodate both the shapes of the `input` and the `indices`. By the way, another way to index into `abcd` is starred-tuple unpacking:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "04010fac",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "assert np.array_equal(abcd[:, *(0, 0), :], a)\n",
+    "assert np.array_equal(abcd[:, *(0, 1), :], b)\n",
+    "assert np.array_equal(abcd[:, *(1, 0), :], c)\n",
+    "assert np.array_equal(abcd[:, *(1, 1), :], d)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "61ef3393",
+   "metadata": {},
+   "source": [
+    "It's also important to understand what `abcd` is; while the original array has dimensions $2 \\times 30 \\times 4$ while `abcd` has distinct (different from `input`) interior dimensions. $2 \\times 2$ seems somehow smaller/related to $30$ but it's not; the indexing dimensions determine output side _at those dimensions_:"
```

Comment:
Again: the dimension given in the text doesn't seem to fit that used in the code (2 vs 3).